### ISSUE TITLE
Issues with type catching in pylance on latest version

### ISSUE THREAD

**Issue by sayyid5416** [microsoft/pylance-release#2726]
<h3> Environment data </h3>

-   Language Server version: 2022.4.2
-   OS and version: win32 x64
-   Python version (and distribution if applicable, e.g. Anaconda): 3.10.4
-   python.analysis.indexing: null
-   python.analysis.typeCheckingMode: basic

<h3> Main Issue </h3>

```python
def test_fctn(arg :str = None):
    if arg:
        print(arg)
```
1. In the above function, On earlier versions of pylance, there was no warning shown for `def test_fctn(arg :str = None):`, while in this new pylance version, a warning is shown like **_Type "None" cannot be assigned to type "str"Pylance[reportGeneralTypeIssues]_**
2. In `if arg:`, arg showed as `str | None` on earlier versions, while on latest version, it just shows of type `str`.

Basically, In earlier versions, In `def test_fctn(arg :str = None):`, pylance automatically catches that `arg` is of type `str | None`, But on latest version, it just tells its of type `str` (Not catching the `None`)

**Comment by erictraut** [comment]
This is intentional. The original draft of PEP 484 indicated that type checkers could infer the type of a function parameter as `Optional` if it was assigned a default argument of `None`. The PEP was later amended to indicate that this was no longer recommended — that the type annotation for the parameter should be specified explicitly as `Optional[str]` (or `str | None`) in this case. Pyright (the type checker that underlies pylance) switched its default behavior to be consistent with the newer guidance in PEP 484. Here's what the updated PEP says:

> A past version of this PEP allowed type checkers to assume an optional type when the default value is None, as in this code: `def handle_employee(e: Employee = None): ...`. This would have been treated as equivalent to: `def handle_employee(e: Optional[Employee] = None) -> None: ...`. This is no longer the recommended behavior. Type checkers should move towards requiring the optional type to be made explicit.

I encourage you to update to the newer standard and make the type explicit in your code. However, if you want the old behavior, you can configure pylance by creating a `pylanceconfig.json` file at the root of your project and including the following:
```
{
    "strictParameterNoneValue": false
}
```

For more details about `pyrightconfig.json` and configuration options, refer to [this documentation](https://github.com/microsoft/pyright/blob/main/docs/configuration.md#type-check-diagnostics-settings).

**Comment by sayyid5416** [comment]
> If you want the old behavior, you can configure pylance by creating a pylanceconfig.json file at the root of your project
VS code is not letting me create file with name `pylanceconfig.json`

**Comment by sayyid5416** [comment]
Thank you @erictraut for the help, Since this is the new intended behavior, I'm closing this issue.
